craft_parts/utils/maven/common.py

```diff
@@ -559,3 +551,74 @@ def _format_xml_str(element: ET.Element) -> str:
     """Get a nicely-formatted string for displaying an XML element."""
     ET.indent(element)
     return ET.tostring(element).decode(errors="replace")
+
+
+def _get_namespaces(project: ET.Element) -> Namespaces:
+    """Find and register the first XML namespace."""
+    namespace = re.search("{(.*)}", project.tag)
+    # Older pom files may not contain a namespace
+    namespaces = {"": namespace.group(1)} if namespace else {}
+    for prefix, uri in namespaces.items():
+        ET.register_namespace(prefix, uri)
+    return namespaces
+
+
+def _get_poms(part_info: PartInfo, existing: GroupDict) -> list[Path]:
+    """Get a list of poms on a project.
+
+    Each submodule is added to the list of existing artifacts because Maven's build
+    will put them into the backstage as they are built. If a build somehow fails
```

Comment:
I don't think this is correct: the modules will be deployed to the **export** dir (and then migrated to the backstage) in the case of the maven-use plugin, but that's not the only place we deploy to currently. Might be simpler to say that the submodules are visible to maven during the build itself, or something like that.

Also, are you sure they are deployed "as they are built"?